platform/windows/os_windows.cpp

```diff
@@ -2260,6 +2260,26 @@ String OS_Windows::get_processor_name() const {
 	}
 }
 
+String OS_Windows::get_unique_id() const {
+	HW_PROFILE_INFOA HwProfInfo;
+	ERR_FAIL_COND_V(!GetCurrentHwProfileA(&HwProfInfo), "");
+	
+	// Create a string from the raw GUID data first
+	String guid_str = String(HwProfInfo.szHwProfileGuid);
+	
+	// Remove any potential null characters within the string
```

Comment:
The string constructor skips any null characters, so this shouldn't be needed, I think just removing the length part of the original is the proper solution